spec.html

```diff
@@ -38326,6 +38326,126 @@ <h1>`Promise.all` Resolve Element Functions</h1>
+              1. Let _steps_ be the algorithm steps defined in <emu-xref href="#sec-promise.allsettled-resolve-element-functions" title></emu-xref>.
+              1. Let _resolveElement_ be ! CreateBuiltinFunction(_steps_, &laquo; [[AlreadyCalled]], [[Index]], [[Values]], [[Capability]], [[RemainingElements]] &raquo;).
+              1. Let _alreadyCalled_ be a new Record { [[Value]]: *false* }.
+              1. Set _resolveElement_.[[AlreadyCalled]] to _alreadyCalled_.
+              1. Set _resolveElement_.[[Index]] to _index_.
+              1. Set _resolveElement_.[[Values]] to _values_.
+              1. Set _resolveElement_.[[Capability]] to _resultCapability_.
+              1. Set _resolveElement_.[[RemainingElements]] to _remainingElementsCount_.
+              1. Let _rejectSteps_ be the algorithm steps defined in <emu-xref href="#sec-promise.allsettled-reject-element-functions" title></emu-xref>.
+              1. Let _rejectElement_ be ! CreateBuiltinFunction(_rejectSteps_, &laquo; [[AlreadyCalled]], [[Index]], [[Values]], [[Capability]], [[RemainingElements]] &raquo;).
+              1. Set _rejectElement_.[[AlreadyCalled]] to _alreadyCalled_.
+              1. Set _rejectElement_.[[Index]] to _index_.
+              1. Set _rejectElement_.[[Values]] to _values_.
+              1. Set _rejectElement_.[[Capability]] to _resultCapability_.
+              1. Set _rejectElement_.[[RemainingElements]] to _remainingElementsCount_.
+              1. Set _remainingElementsCount_.[[Value]] to _remainingElementsCount_.[[Value]] + 1.
+              1. Perform ? Invoke(_nextPromise_, `"then"`, &laquo; _resolveElement_, _rejectElement_ &raquo;).
+              1. Increase _index_ by 1.
+          </emu-alg>
+        </emu-clause>
+    
+        <emu-clause id="sec-promise.allsettled-resolve-element-functions">
+          <h1>`Promise.allSettled` Resolve Element Functions</h1>
+          <p>A `Promise.allSettled` resolve element function is an anonymous built-in function that is used to resolve a specific `Promise.allSettled` element. Each `Promise.allSettled` resolve element function has [[Index]], [[Values]], [[Capability]], [[RemainingElements]], and [[AlreadyCalled]] internal slots.</p>
+          <p>When a `Promise.allSettled` resolve element function is called with argument _x_, the following steps are taken:</p>
+          <emu-alg>
+            1. Let _F_ be the active function object.
+            1. Let _alreadyCalled_ be _F_.[[AlreadyCalled]].
+            1. If _alreadyCalled_.[[Value]] is *true*, return *undefined*.
+            1. Set _alreadyCalled_.[[Value]] to *true*.
+            1. Let _index_ be _F_.[[Index]].
+            1. Let _values_ be _F_.[[Values]].
+            1. Let _promiseCapability_ be _F_.[[Capability]].
+            1. Let _remainingElementsCount_ be _F_.[[RemainingElements]].
+            1. Let _obj_ be ! ObjectCreate(%ObjectPrototype%).
```

Comment:
would it be worth (for readability) making an abstract operation here (and below) like `CreateSettledResult(true/false, x)`?

Reply:
That sounds like a potentially beneficial refactoring! Would you like this to happen as part of this PR, or separately as a follow-up?

Reply:
I guess it could be either, but it’d be nice to do it as part of this PR :-)

Reply:
it seems with that change we could also fold these two functions into one